src/dex-ui/pages/dao/MultiSigDAODashboard/TransactionsList.tsx

```diff
@@ -13,12 +13,28 @@ const transactionTabs = [
 ];
 
 export function TransactionsList() {
-  const { dao } = useOutletContext<MultiSigDAODetailsContext>();
-  const { accountId: daoAccountId, safeId: safeAccountId } = dao;
+  const { dao } = useOutletContext<DAODetailsContext>();
+  const { accountId: daoAccountId } = dao;
+  let safeAccountId = "";
+  let tokenId = "";
+  let governanceAddress = "";
+  if (dao.type === DAOType.MultiSig) {
+    safeAccountId = dao.safeId;
+  } else if (dao.type === DAOType.GovernanceToken) {
+    tokenId = dao.tokenId;
+    governanceAddress = dao.governanceAddress;
+  }
```

Comment:
we can directly typecast the required values like `const {safeId: safeAccountId  } =  dao as MultiSigDetails;` in other dao's cases it would be undefined which can be used to control the UI and Hooks behaviour.

Reply:
Added a TODO here for the time. will pick it up after finishing proposal details flow.